lib/ansible/modules/cloud/amazon/rds_instance.py

```diff
@@ -0,0 +1,843 @@
+            del(params['DBSnapshotIdentifier'])
+        else:
+            params["SkipFinalSnapshot"] = True
+        response = conn.delete_db_instance(**params)
+        instance = RDSDBInstance(response['DBInstance'])
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg="Failed to delete instance: %s " % str(e),
+                         exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.message) )
+
+    # If we're not waiting for a delete to complete then we're all done
+    # so just return
+    if not module.params.get('wait'):
+        module.exit_json(changed=True, operation="delete")
+    try:
+        instance = await_resource(conn, instance, 'deleted', module)
+        module.exit_json(changed=True)
+    except botocore.exceptions.ClientError as e:
+        if e.code == 'DBInstanceNotFound':
+            module.exit_json(changed=True, operation="delete")
+        else:
+            module.fail_json(msg="Failure waiting for deletion to complete: %s " % str(e),
+                             exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.message) )
+    assert(False), "error in module logic - this should not be reached"
+
+def update_rds_tags(module, conn, resource, current_tags, desired_tags):
+    changed = False
+    # it's much easier to do set manipulation in ansible form
+    current_tags = boto3_tag_list_to_ansible_dict(current_tags)
+    desired_tags = boto3_tag_list_to_ansible_dict(desired_tags)
+    current_keys = set(current_tags.keys())
+    desired_keys = set(desired_tags.keys())
+    to_add = desired_keys - current_keys
+    to_delete = current_keys - desired_keys
+
+    for k in desired_keys & current_keys:
+        if current_tags[k] != desired_tags[k]:
+            to_delete.add(k)
+            to_add.add(k)
+
+    if to_delete:
+        try:
+            conn.remove_tags_from_resource(ResourceName=resource, TagKeys=list(to_delete))
+            changed = True
+        except botocore.exceptions.ClientError as e:
+            module.fail_json(msg="Failure removing instance tags: %s" % str(e),
+                             exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.message) )
+    if to_add:
+        try:
+            conn.add_tags_to_resource(ResourceName=resource,
+                                      Tags=[{'Key': k, 'Value': desired_tags[k]}
+                                            for k in to_add])
+            changed = True
+        except botocore.exceptions.ClientError as e:
+            module.fail_json(msg="Failure adding instance tags: %s" % str(e),
+                             exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.message) )
+    return changed
+
+
+def modify_db_instance(module, conn):
+    force_password_update = module.params.pop('force_password_update')
+    required_vars = ['instance_name']
+    valid_vars = ['apply_immediately', 'backup_retention', 'backup_window',
```

Comment:
I keep having problems that variables are defined for create but are not defined for modify (or occasionally the opposite way).  Normally I will create an RDS once and then use the same definition later to verify it, so I think that exactly the same parameters should be valid for both.  

a) propose sharing the "valid_vars" list between the two functions.  

b) once we are doing that, maybe the entry logic of the two functions should be merged?  

What do you think about this?